src/Analyzer.Cli/CommandLineParser.cs

```diff
@@ -243,23 +243,69 @@ private void FindJsonFilesInDirectoryRecursive(DirectoryInfo directoryPath, List
             }
         }
 
-        private bool IsValidSchema(string template)
+        private static bool IsValidTemplate(string template)
         {
-            JObject jsonTemplate = JObject.Parse(template);
-            string schema = (string)jsonTemplate["$schema"];
-            string[] validSchemas = { 
-                "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
-                "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"};
-            return validSchemas.Contains(schema);
+            var reader = new JsonTextReader(new StringReader(template));
+
+            reader.Read();
+            if (reader.TokenType != JsonToken.StartObject)
+            {
+                return false;
+            }
+
+            // Checks all top level properties for schema property. If schema is not found, will check property names against expected ARM template properties. If expected properties
+            // are not found function will return false.
+
+            string[] validTemplateProperties =
+            {
+                 "contentVersion",
+                 "apiProfile",
+                 "parameters",
+                 "variables",
+                 "functions",
+                 "resources",
+                 "outputs",
+            };
+
+            string[] validSchemas = {
+               "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+               "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+               "https://schema.management.azure.com/schemas/2018-05-01/subscriptionDeploymentTemplate.json#",
+               "https://schema.management.azure.com/schemas/2019-08-01/tenantDeploymentTemplate.json#",
+               "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#"
+            };
+
+            while (reader.Read())
+            {
+                if (reader.Depth == 1 && reader.TokenType == JsonToken.PropertyName)
+                {
+
+
+                    if (reader.Value.ToString() == "$schema")
+                    {
+                        reader.Read();
+
+                        if (reader.TokenType != JsonToken.String)
+                        {
+                            return false;
+                        }
+
+                        return validSchemas.Any(property => JsonToken.Equals(reader.Value,StringComparison.OrdinalIgnoreCase));
```

Comment:
I don't think this is done right - it probably doesn't compile.  Please double-check this after resolving the conflict.